src/stores/ResourceConfig.tsx

```diff
@@ -364,14 +523,21 @@ const getInitialState = (
             if (data && data.length > 0) {
                 const { setResourceConfig, preFillCollections } = get();
 
-                data[0].spec.bindings.forEach((binding: any) =>
-                    setResourceConfig(binding.source, {
+                const collectionNameProp =
+                    entityType === ENTITY.MATERIALIZATION ? 'source' : 'target';
```

Comment:
We should potentially move this to a stand alone function... could wait until we need it used elsewhere.

But would be good to try to start keeping as much of the difference between entities in a single location if possible.

Reply:
It is more suitable to address this in the PR to combine the create and edit workflows. Not a necessary fix now.